lib/stud/interval.rb

```diff
@@ -1,38 +1,83 @@
-require "stud/task"
+require "thread"
+
 module Stud
+  STUD_STOP_REQUESTED = :stud_stop_requested
+
+  extend self
```

Comment:
This feels like it pollutes the intent of the module.

Extending self makes `Stud.interval` available as well as making `class Foo; include Stud; end` provide `Foo.new.interval` which is ... probably not what we want?

The Stud module here only provides a namespace for these methods (Stud.interval) and I don't think we  want the the ability to `include Stud` from a class. Thoughts?